addons/sms/models/sms_sms.py

```diff
@@ -16,74 +17,72 @@ class SmsSms(models.Model):
     _order = 'id DESC'
 
     IAP_TO_SMS_STATE = {
+        'processing': 'process',
         'success': 'sent',
+        'delivered': 'delivered',  # not used for _postprocess_iap_results but webhook events.
         'insufficient_credit': 'sms_credit',
         'wrong_number_format': 'sms_number_format',
+        'country_not_supported': 'sms_country_not_supported',
         'server_error': 'sms_server',
         'unregistered': 'sms_acc'
     }
 
+    BOUNCE_DLR_ERRORS = {'sms_invalid_destination', 'sms_not_allowed', 'sms_rejected'}
+    DLR_ERRORS = {'sms_expired', 'sms_not_delivered', *BOUNCE_DLR_ERRORS}
+
+    uuid = fields.Char(
+        'UUID', help='UUID used by sms service', copy=False, readonly=True, index=True, default=lambda __: uuid4().hex,
```

Comment:
"Alternate way to identify a SMS record, used to tie in with SMS delivery reports".

(Also, others seem written as: `default=lambda self: str(uuid.uuid4())`, don't really like the double underscore).